sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/JacksonVersion.java

```diff
@@ -0,0 +1,108 @@
+ */
+final class JacksonVersion {
+    private SemanticVersion annotationsVersion;
+    private SemanticVersion coreVersion;
+    private SemanticVersion databindVersion;
+    private SemanticVersion xmlVersion;
+    private SemanticVersion jsr310Version;
+
+    private static final String ANNOTATIONS_PACKAGE_NAME = "jackson-annotations";
+    private static final String CORE_PACKAGE_NAME = "jackson-core";
+    private static final String DATABIND_PACKAGE_NAME = "jackson-databind";
+    private static final String XML_PACKAGE_NAME = "jackson-dataformat-xml";
+    private static final String JSR310_PACKAGE_NAME = "jackson-datatype-jsr310";
+
+    private static final SemanticVersion MIN_SUPPORTED_VERSION = SemanticVersion.parse("2.10.0");
+    private static final SemanticVersion MAX_SUPPORTED_VERSION = SemanticVersion.parse("2.13.0-rc1");
+
+    private static final String AZURE_CORE_VERSION = CoreUtils
+        .getProperties("azure-core.properties")
+        .getOrDefault("version", SemanticVersion.UNKNOWN_VERSION);
+
+    private final String helpString;
+    private final boolean criticalErrorDetected;
+    private final ClientLogger logger = new ClientLogger(JacksonVersion.class);
+
+    private JacksonVersion() {
+        annotationsVersion = getVersion("com.fasterxml.jackson.annotation.JsonProperty");
+        coreVersion = getVersion("com.fasterxml.jackson.core.JsonGenerator");
+        databindVersion = getVersion("com.fasterxml.jackson.databind.ObjectMapper");
+        xmlVersion = getVersion("com.fasterxml.jackson.dataformat.xml.XmlMapper");
+        jsr310Version = getVersion("com.fasterxml.jackson.datatype.jsr310.JavaTimeModule");
+        criticalErrorDetected = checkVersion(annotationsVersion, ANNOTATIONS_PACKAGE_NAME)
+            | checkVersion(coreVersion, CORE_PACKAGE_NAME)
+            | checkVersion(databindVersion, DATABIND_PACKAGE_NAME)
+            | checkVersion(xmlVersion, XML_PACKAGE_NAME)
+            | checkVersion(jsr310Version, JSR310_PACKAGE_NAME);
+        helpString = formatHelpString();
+        logger.info(helpString);
+    }
+
+    public String getHelpInfo() {
+        return helpString;
+    }
+
+    private SemanticVersion getVersion(String className) {
+        try {
+            return SemanticVersion.getPackageVersionForClass(className);
+        } catch (Throwable e) {
+            logger.warning("Failed to retrieve package version for class " + className, e);
+            return SemanticVersion.createInvalid();
+        }
+    }
+
+    private static JacksonVersion instance = null;
+    public static synchronized JacksonVersion getInstance() {
+        if (instance == null) {
+            instance = new JacksonVersion();
+        }
+
+        return instance;
+    }
+
+    private boolean checkVersion(SemanticVersion version, String packageName) {
+        if (!version.isValid()) {
+            logger.warning("Could not find version of '{}'.", packageName);
+            return false;
+        }
```

Comment:
Here we were not able to get version (e.g. it may happen when you debug locally), i.e. we can't conclusively say if this version is compatible or not, i.e. we don't indicate the critical error.
Maybe since the critical error is just a flag on the log, I can remove it - by looking at versions we should be able to educe it ourselves.